definitions/npm/styled-components_v4.x.x/flow_v0.95.2-/styled-components_v4.x.x.js

```diff
@@ -0,0 +1,444 @@
+// @flow
+
+declare module 'styled-components' {
+  // Could this use `React$ElementRef<>` instead?
+  // It might be more correct and future-proof
+  // Since it won't rely on us keeping this up to date
+  // TODO: Experiment/Look into this
```

Comment:
In most cases, I don't believe using `React$ElementRef<>` will make a practical difference, as (for example) `React$ElementRef<'div'>` will return `HTMLDivElement`, which is what is already listed here. To your point about future-proofing these types, though, I could see a benefit for using `React$ElementRef`, in case some of the elements that don't have specialized element types gain those in the future (I imagine the chances of that are rare, but possible?). Given that, I'd say +1 for using `React$ElementRef`!

Reply:
I am not sure which one of these two we should do then, though:

 1.  `div: 'div'` and then use the `V` inside `React$ElementRef` when building `ConvenientShorthands`. This is slightly more "convenient" in case Flow devs decide to change their mind (wouldn't be the first time), but it's slightly less obvious WTF is going on.

 2.  `div: React$ElementRef<'div'>`, and just use that as `V` when building `ConvenientShorthands`. This has the benefit of being a little bit more explicit, but it might require more "work"<sup>[1](#workbitch)</sup> in the future to keep up

----

<a name="workbitch">1:</a> "Work": Well, it will probably just be running `sed` to search & replace stuff

Reply:
Given that these defs will be community maintained by several different people, the clarity offered by option 2 may be worth the extra work. Especially considering that extra work could be automated, as you mentioned.

Reply:
> Given that these defs will be community maintained by several different people, the clarity offered by option 2 may be worth the extra work. Especially considering that extra work could be automated, as you mentioned.

Yes, that's a very good point.

Reply:
I'll mark this as resolved, as we have https://github.com/omninonsense/flow-typed/pull/1